WordPress/Classes/ViewRelated/Post/PostSettingsViewController+JetpackSocial.swift

```diff
@@ -3,39 +3,78 @@ extension PostSettingsViewController {
 
     @objc func showNoConnection() -> Bool {
         let isJetpackSocialEnabled = FeatureFlag.jetpackSocial.enabled
+        let isNoConnectionViewHidden = UserPersistentStoreFactory.instance().bool(forKey: hideNoConnectionViewKey())
         let blogSupportsPublicize = apost.blog.supportsPublicize()
         let blogHasNoConnections = publicizeConnections.count == 0
         let blogHasServices = availableServices().count > 0
 
         return isJetpackSocialEnabled
+        && !isNoConnectionViewHidden
         && blogSupportsPublicize
         && blogHasNoConnections
         && blogHasServices
     }
 
     @objc func createNoConnectionView() -> UIView {
         let services = availableServices()
-        let viewModel = JetpackSocialNoConnectionViewModel(
-            services: services,
-            onConnectTap: {
-                // TODO: Open the social screen
-                print("Connect tap")
-            }, onNotNowTap: {
-                // TODO: Add condition to hide the connection view after not now is tapped
-                print("Not now tap")
-            }
-        )
+        let viewModel = JetpackSocialNoConnectionViewModel(services: services,
+                                                           onConnectTap: onConnectTap(),
+                                                           onNotNowTap: onNotNowTap())
         let viewController = JetpackSocialNoConnectionView.createHostController(with: viewModel)
 
         // Returning just the view means the view controller will deallocate but we don't need a
         // reference to it. The view itself holds onto the view model.
         return viewController.view
     }
 
+    private func hideNoConnectionViewKey() -> String {
+        guard let dotComID = apost.blog.dotComID?.stringValue else {
+            return Constants.hideNoConnectionViewKey
+        }
+
+        return "\(dotComID)-\(Constants.hideNoConnectionViewKey)"
```

Comment:
I feel like having a site-specific preference class would be helpful, seeing that we have so many use cases for it 😅